src/TeamSpeak3QueryApi/Specialized/ITypeCaster.cs

```diff
@@ -87,4 +87,24 @@ public dynamic Cast(object source)
             return TimeSpan.FromSeconds(int.Parse(source.ToString()));
         }
     }
+
+    class DateTimeTypeCaster : ITypeCaster
+    {
+        public dynamic Cast(object source)
+        {
+            if (source == null)
+                return false;
+
+            long unixTimestamp;
+
+            if (source is long)
+                unixTimestamp = (long)source;
+            else
+                unixTimestamp = long.Parse(source.ToString());
+
+            var timeSpan = TimeSpan.FromSeconds(unixTimestamp);
+            var epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+            return epoch.Add(timeSpan);
```

Comment:
Could we use the [builtin function](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimeseconds?view=netstandard-1.3) for this?

Reply:
Totally forgot about DateTimeOffset